### DISCUSSION TITLE
Sensori di movimento, status codes e rilevamento

### DISCUSSION THREAD

**Discussion by maxill1** [maxill1/ialarm-mqtt#42] [General]
Mi ricollego al [dubbio](https://github.com/maxill1/ialarm-mqtt/issues/20#issuecomment-1323520234) di @blakdog1978 relativo ai sensori di movimento per capire se la gestione messa in piedi è migliorabile e soprattutto spiegarvi quali limiti tecnici abbiamo davanti.

Partiamo da alcune premesse doverosa basata sulla mia personale esperienza:
- Quando un sensore si triggera da centrale **disarmata** sul display potete vedere **F-01, F-02**, etc dove F è sinonimo di "Fault" (ma inteso come "sensore aperto o tamperato") e il numero 01, 02 è la zona.
- Quando un sensore si triggera da centrale **armata** sul display potete vedere **A-01, A-02**, etc dove A è sinonimo di "Alarm" e il numero 01, 02 è la zona.

I codici F-xx si resettano da soli quando il sensore torna allo stato "OK/chiuso" i codici A-xx si resettano quando si fa il clear degli errori (si disarma la centrale due volte oppure si fa clear da applicazione android).

Ma F-xx non vuol dire solo "aperto", può voler dire errore di connessione, tamper, etc...e questo lo si può leggere nella proprietà "message" che ogni sensore espone su ialarm-mqtt (decodificata qui:  https://github.com/maxill1/node-ialarm/blob/master/src/meian-status-decoder.js)

io ho testato:
- Meian MC-8250 (venduti da antifurto365) che non riportano mai F-xx ma solo A-xx (quindi a centrale disarmata non ho avuto fortuna a intercettare movimento).
- Optex BX-80N wireless (venditi da antifurto365) che riportano sia F-xx che A-xx

Immagino che molto dipenda da come sono monitorati dalla centrale (e come sono configurati i jumper!?)... quindi solito discorso se la centrale vede il sensore aperto scrivendo sul display F-01, F-03, etc allora lo vede anche ialarm-mqtt, altrimenti no.

> Ciao mi rimane un problema sui sensori di movimento wifi, ma non sono sicuro di dove aprire issue, più o meno il problema e che non credo fuzionino come dovrebbero, ho provato ad abilitarne due su sei , il primo gli ho cambiato le batterie e messo in funzione, il secondo l'ho smontato tolto le batterie dunque è sucuramente disconnesso dalla centrale allego due immagini dei sensori la prima di quello che dovrebbe funzionare e la seconda di quello smontato. P.S. Ho due sensori wifi su due velux e quelli funzionano bene mi danno problemi questi di movimento. ![zone 1](https://user-images.githubusercontent.com/36457031/203301301-4a9296a8-dee9-4135-9f9f-ff1ef7fe5113.JPG) ![Zone 2](https://user-images.githubusercontent.com/36457031/203301323-6e9f8fdd-b2c6-440b-9ca5-df92b728f143.JPG)

Nel tuo caso specifico:

Lo **Status=9** che riporta uno dei tuoi sensori è un rilevamento "**aperto**" o "**movimento**" (dipende da che sensore è) e mi riporti però che lo Status non torna mai a OK (Status=1). 

Potrebbe trattarsi si un problema di cache di ialarm-mqtt (anche se credo di averli risolti), ma potrebbe anche trattarsi di un tipo di sensore "**normalmente aperto**", cioè che si triggera solo quando riporta chiusura del contatto (e mai in apertura).

Io ne ho uno cosi per l'allagamento che mi da sempre Fault con Status=9.  L'ho configurato aggiungendo alla zona numero 16 il contactType ( "zones" sullo yaml) che inverte la logica di "Fault".

![image](https://user-images.githubusercontent.com/367042/203369089-408badcd-ac47-4b42-946b-bb49ec39cdc0.png)

Nell'esempio che segue ho impostato quanto descritto insieme ad altri sensori (ai soli fini di classificazione su HA): 
-  un sensore porta finestra come **door** (sempre per farlo vedere a HA con la porticina)
- i due Meian MC-8250 (zona 12 e 13) come **motion** (per farli vedere a HA con il simbolo dell'omino)
- un "normalmente aperto" che torna sempre status 9 (allagamento)

```
zones: 
  - number: 1
    device_class: door
  - number: 12
    device_class: motion
  - number: 13
    device_class: motion
  - number: 16
    contactType: NO
    device_class: moisture
    statusProperty: fault
```

L'altro sensore che hai staccato riporta **Status=41** e dire che è disconnesso (Wireless lost) quindi mi sembra corretto. Sbaglio?

**Comment by blakdog1978** [comment]
È corretto provo ad aggiungere la configurazione da te suggerita poi ti aggiorno, grazie

**Comment by blakdog1978** [comment]
Come sempre non posso fare altro che ringraziarti, è bastato configurarli NO e hanno cominciato a funzionare.

**Reply by Eckart0** [reply]
Salve,

due punti:
1)
Ho identificato gli stati "1:ok", "5:Bypass" e "9:open". Esiste un elenco degli altri codici numerici?
2)
Non potrebbe esserci una conversione in testo semplice nell'AddOn - come attributo di un sensore o meglio come sensore aggiuntivo?
Ora mi sto accontentando di un'entità modello:
```- platform: template
  sensors:
    status_alarmzone_33:
      friendly_name: "Zone 33: Haustür UG"
      value_template: >-
        {% if (state_attr('binary_sensor.alarm_zone_33_haustuer_ug', 'status') == 1 ) %}
          ok
        {% elif (state_attr('binary_sensor.alarm_zone_33_haustuer_ug', 'status') == 9) %}
          offen
        {% else %}
          {{ state_attr('binary_sensor.alarm_zone_33_haustuer_ug', 'status') }}
        {% endif %}`
```
Saluti, Eckart

**Comment by maxill1** [comment]
Domanda per chi altro ha i sensori porta/finestra MD-210R, a voi riportano lo stato della batteria alla centrale? 

Mi riferisco puramente alla centrale e app dedicata (non a ialarm-mqtt).

A me quando si scaricano riportano nel pannello web solo "wireless sensor loss" e non "battery low"...

**Reply by Nik71git** [reply]
non sono certo di avere gli stessi sensori ma probabilmente sì. Nella centrale vedo solo il Fault del sensore, il bypass eventuale o se ha innescato l'allarme.
